### PR TITLE
Fix crash in tournament client when pick/ban is activated

### LINKED ISSUE #24589: `UnmaskedTournamentBeatmapPanel` causes crash in tournament client
### Type

Crash to desktop

### Bug description

`UnmaskedTournamentBeatmapPanel` (introduced in https://github.com/ppy/osu/pull/24565) is causing the lazer tournament client to crash with the following stack trace:
```
System.InvalidOperationException: Can not have border effects/edge effects if masking is disabled.
   at osu.Framework.Graphics.Containers.CompositeDrawable.CompositeDrawableDrawNode.ApplyState()
   at osu.Framework.Graphics.Drawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Graphics.Containers.CompositeDrawable.addFromComposite(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode, Int32& j, CompositeDrawable parentComposite, List`1 target)
   at osu.Framework.Graphics.Containers.CompositeDrawable.GenerateDrawNodeSubtree(UInt64 frame, Int32 treeIndex, Boolean forceNewDrawNode)
   at osu.Framework.Platform.GameHost.UpdateFrame()
   at osu.Framework.Threading.GameThread.processFrame()
--- End of stack trace from previous location ---
   at osu.Framework.Platform.GameHost.<>c__DisplayClass139_0.<abortExecutionFromException>b__0()
   at osu.Framework.Threading.ScheduledDelegate.InvokeTask()
   at osu.Framework.Threading.ScheduledDelegate.RunTaskInternal()
   at osu.Framework.Threading.Scheduler.Update()
   at osu.Framework.Threading.GameThread.processFrame()
   at osu.Framework.Threading.GameThread.RunSingleFrame()
   at osu.Framework.Platform.ThreadRunner.RunMainLoop()
   at osu.Framework.Platform.GameHost.windowUpdate()
   at osu.Framework.Platform.SDL2Window.RunFrame()
   at osu.Framework.Platform.SDL2Window.RunMainLoop()
   at osu.Framework.Platform.SDL2Window.Run()
   at osu.Framework.Platform.GameHost.Run(Game game)
   at osu.Desktop.Program.Main(String[] args) in /Users/daohe/RiderProjects/osu/osu.Desktop/Program.cs:line 118
```

I've attached a [`bracket.json` as well as an `ipc.txt` in this .zip](https://github.com/ppy/osu/files/12385395/bracketjson_plus_ipc.zip) for reproducing, if necessary. Steps to reproduce: 
- go to mappool screen
- pick DT4
- go to gameplay screen

### Screenshots or videos

https://github.com/ppy/osu/assets/1176059/306b09dc-fa0f-4013-b1bf-9a7313f782dc

### Version

master at revision 4ee7fbd9b0c539824727a7a06a3f80e69a44f571

### Logs

[network.log](https://github.com/ppy/osu/files/12385404/network.log)
[performance.log](https://github.com/ppy/osu/files/12385405/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12385406/runtime.log)

### PR DESCRIPTION
**Pull request by peppy**
Turns out the masking I added breaks when a border style is applied for picks/bans, and it wasn't doing much for visuals anyway ( was mostly there for the rounded corners which are not even featured in the tournament client yet.

Closes https://github.com/ppy/osu/issues/24589.

### CODE REVIEW

File: osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs [hunk 1]
```diff
@@ -38,12 +45,14 @@ public void TestSongBar()
...
             {
                 var beatmap = CreateAPIBeatmap(Ruleset.Value);
+
                 beatmap.CircleSize = 3.4f;
                 beatmap.ApproachRate = 6.8f;
                 beatmap.OverallDifficulty = 5.5f;
                 beatmap.StarRating = 4.56f;
                 beatmap.Length = 123456;
                 beatmap.BPM = 133;
+                beatmap.OnlineID = CreateSampleBeatmap().OnlineID;
```
Comment (bdach) [inline osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs:55]: The test doesn't cover the failure. When I check out https://github.com/ppy/osu/pull/24601/commits/5009fd379421f928bca35028b00e6b88374bdb5e, there's no crash.

There _is_ one if I do this:

```diff
diff --git a/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs b/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs
index d52b453185..e0444b6126 100644
--- a/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs
+++ b/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs
@@ -14,6 +14,7 @@ namespace osu.Game.Tournament.Tests.Components
     public partial class TestSceneSongBar : TournamentTestScene
     {
         private SongBar songBar = null!;
+        private TournamentBeatmap ladderBeatmap = null!;
 
         [SetUpSteps]
         public override void SetUpSteps()
@@ -22,9 +23,10 @@ public override void SetUpSteps()
 
             AddStep("setup picks bans", () =>
             {
+                ladderBeatmap = CreateSampleBeatmap();
                 Ladder.CurrentMatch.Value!.PicksBans.Add(new BeatmapChoice
                 {
-                    BeatmapID = CreateSampleBeatmap().OnlineID,
+                    BeatmapID = ladderBeatmap.OnlineID,
                     Team = TeamColour.Red,
                     Type = ChoiceType.Pick,
                 });
@@ -52,7 +54,7 @@ public void TestSongBar()
                 beatmap.StarRating = 4.56f;
                 beatmap.Length = 123456;
                 beatmap.BPM = 133;
-                beatmap.OnlineID = CreateSampleBeatmap().OnlineID;
+                beatmap.OnlineID = ladderBeatmap.OnlineID;
 
                 songBar.Beatmap = new TournamentBeatmap(beatmap);
             });

```

tl;dr: `CreateSampleBeatmap()` doesn't return the same `OnlineID` every time.

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs:55]: Ugh, that's a final change I made to fix the online ID always being 1. Always gets me (refactoring after confirming coverage).

Your change looks more correct.

**Review by bdach — APPROVED** [review]
